### ISSUE TITLE
Doesn't work with blur() modifier

### ISSUE THREAD

**Issue by dqhieu** [gualtierofrigerio/ScreenshotSwiftUI#5]
So I have a view like this

```swift
ZStack {
  Circle()
    .fill(Color.blue.opacity(0.5))
    .blur(radius: 150)
  MyView()
}
```

Here's the expected screenshot, but I'm getting this screenshot instead

|Expectation|Actual|
|-|-|
|![expected](https://github.com/gualtierofrigerio/ScreenshotSwiftUI/assets/8291457/ea6f0469-b225-4e8f-a5c9-33489d97b136)|![actual](https://github.com/gualtierofrigerio/ScreenshotSwiftUI/assets/8291457/82751f55-a5e7-4e1a-b1d3-e028e91f9829)|

So basically when taking screenshot it doesn't render the `.blur()` modifier of the `Circle`

Do you have any idea how to fix it?

Thanks for checking

**Comment by gualtierofrigerio** [comment]
from your screenshot, it looks like it is also ignoring the fill modifier as the color looks different.
I suppose the text is in MyView so it looks like is doing a screenshot of the uppermost view, which could be possible. 
More than blur, I guess It is an issue with ZStack, if you apply blur modifier to another view outside a ZStack does it work?

**Comment by dqhieu** [comment]
So I tried `blur()` without using `ZStack`, and the result is still the same. So I think it's not relevant to `ZStack`

```swift
struct ContentView: View {
  
  @State var screenshotMaker: ScreenshotMaker?
  
  var body: some View {
    VStack {
      Image(systemName: "globe")
        .imageScale(.large)
        .foregroundStyle(.tint)
      Text("Hello, world!")
      Button(action: {
        guard let uiImage = screenshotMaker?.screenshot() else { return }
        // share image
      }, label: {
        Text("Screenshot")
      })
      Circle()
        .fill(.blue)
        .blur(radius: 50)
    }
    .padding()
    .screenshotMaker { maker in
      screenshotMaker = maker
    }
  }
}
```

|View|Screenshot|
|-|-|
|![Simulator Screenshot - iPhone 14 Pro - 2023-08-02 at 09 42 34](https://github.com/gualtierofrigerio/ScreenshotSwiftUI/assets/8291457/30d4f5cc-9921-4aff-b510-f8dacf5a9f6f)|![screenshot](https://github.com/gualtierofrigerio/ScreenshotSwiftUI/assets/8291457/218dd6f7-495f-4b9d-9832-0a37e9aae4f9)|

**Comment by gualtierofrigerio** [comment]
good catch, it is not working with blur and I guess it is due to how the screenshot is made via UIGraphicsImageRenderer. If I take it with UIGraphicsGetImageFromCurrentImageContext I can see the blur but I'd need to adapt the screenshot function to take only the correct area. Vacation time is near so I don't know if I'll find time to do it before that, but thanks for reporting this

**Comment by dqhieu** [comment]
Sadly UIGraphicsGetImageFromCurrentImageContext will be deprecated in iOS 17

**Comment by gualtierofrigerio** [comment]
oh yes I was opening the project with Xcode 14 so I didn't notice the deprecation message. So anyway this doesn't seem to be an issue related to how SwiftUI renders the blur, but on how the screenshot is made via UIGraphicsImageRenderer
